src/main/java/com/theironyard/controllers/SpringController.java

```diff
@@ -0,0 +1,93 @@
+package com.theironyard.controllers;
+
+import com.theironyard.entities.Chef;
+import com.theironyard.entities.Recipe;
+import com.theironyard.services.ChefRepository;
+import com.theironyard.services.RecipeRepository;
+import com.theironyard.utilities.PasswordStorage;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import javax.annotation.PostConstruct;
+import javax.servlet.http.HttpSession;
+import java.util.List;
+
+@Controller
+public class SpringController {
+    @Autowired
+    RecipeRepository recipes;
+    @Autowired
+    ChefRepository chefs;
+
+    @PostConstruct
+    public void init() throws PasswordStorage.CannotPerformOperationException {
+        if (chefs.count() == 0) {
+            Chef chef = new Chef();
+            chef.setUserName("Amber");
+            chef.setPassword("foodlove11");
+            chefs.save(chef);
+        }
+    }
+
+    @RequestMapping(path = "/", method = RequestMethod.GET)
+    public String home(HttpSession session, Model model, String name, String type) {
+        String userName = (String) session.getAttribute("userName");
+        Chef chef = chefs.findFirstByuserName(userName);
+        if (chef != null) {
+            model.addAttribute("chef", chef);
+
+        }
+        List<Recipe> recipeList = recipes.findAll();
+        if (type != null) {
```

Comment:
I think you may have forgot some code here in this IF statement.  What is this "type" parameter?  Just remove it if you're not using it.